src/lib/grammar.rs

```diff
@@ -235,6 +236,11 @@ impl Grammar {
         self.terminal_precs.get(usize::from(i)).map_or(None, |x| Some(*x))
     }
 
+    /// Return an iterator which produces (in no particular order) all this grammar's valid RIdxs.
```

Comment:
This isn't currently used by lrtable or lrpar, but I think it will be at some point. I will add a test though.